Add documentation for installing plugins in air-gapped environments (#254)
vmware-tanzu/tanzu-cli: docs/full/README.md

publicly available registry to the private registry using the below-mentioned
steps:

1. Download the plugin-inventory image along with all selected plugin images
as a `tar.gz` file on a local disk from the machine which has internet access
using the `tanzu plugin download-bundle` command.
2. Copy this `tar.gz` file to the air-gapped network (using a USB drive or
other mechanism).
3. Upload the plugin bundle `tar.gz` to the air-gapped private registry using
the `tanzu plugin upload-bundle` command.

### Downloading plugin bundle

You can download all plugins within the default central repository by running
the following command:

```sh
tanzu plugin download-bundle --to-tar /tmp/plugin_bundle_complete.tar.gz
```

However, If you want to just migrate plugins within specific plugin groups
(e.g. `vmware-tkg/default:v2.1.0`) you can run the below command to download
the plugin bundle containing only plugins from specified groups:

```sh
tanzu plugin download-bundle --group vmware-tkg/default:v2.1.0 --to-tar /tmp/plugin_bundle_tkg_v2_1_0.tar.gz
```

To migrate plugins from the specific plugin repository and not use the default

> plugin repository you can provide a `--image` tag with the above command. Example:

Reviewer: ```suggestion
plugin repository you can provide a `--image` flag with the above command. Example:
```

```sh
tanzu plugin download-bundle
                  --image custom.repo.example.com/tanzu-cli/plugins/plugin-inventory:latest
                  --group vmware-tkg/default:v2.1.0
                  --to-tar /tmp/plugin_bundle_tkg_v2_1_0.tar.gz
```

### Uploading plugin bundle to the private registry

Once you download the plugin bundle as a `tar.gz` file and copy the file to the
air-gapped network, you can run the following command to migrate plugins to the
private registry (e.g. `registry.example.com/tanzu-cli/plugin`).

```sh
tanzu plugin upload-bundle --tar /tmp/plugin_bundle_complete.tar.gz --to-repo `registry.example.com/tanzu-cli/plugin`
```

> The above-mentioned commands uploads the plugin bundle to the provided private

Reviewer: ```suggestion
The above-mentioned command uploads the plugin bundle to the provided private
```

repository location with the image name `plugin-inventory:latest`. So for the
above example, the plugin inventory image will be published to
`registry.example.com/tanzu-cli/plugin/plugin-inventory:latest`.

> Please note that `tanzu plugin upload-bundle` uploads the plugins by merging

Reviewer: ```suggestion
Please note that `tanzu plugin upload-bundle` uploads the plugins by adding them
```

> the plugin-inventory if exists already. That means if you have already uploaded

Reviewer: ```suggestion
to any plugin-inventory already present in the private registry. That means if you have already uploaded
```

any plugins to the specified private repository, it will keep the existing
plugins and append new plugins from the plugin bundle provided.

You can use this image and configure the default discovery source to point to
this image by running the following command:

```sh
tanzu plugin source update default --uri registry.example.com/tanzu-cli/plugin/plugin-inventory:latest
```

> Now, Tanzu CLI should be able to discover plugins from this newly configured

Reviewer: ```suggestion
Now, the Tanzu CLI should be able to discover plugins from this newly configured
```
